quickstarts/microsoft.compute/vm-cyclecloud/main.bicep

```diff
@@ -0,0 +1,204 @@
+
+resource nsg 'Microsoft.Network/networkSecurityGroups@2020-06-01' = {
+  name: 'cc-nsg'
+  location: location
+  properties: {
+    securityRules: [
+      {
+        name: 'SSH'
+        properties: {
+          priority: 1000
+          protocol: 'Tcp'
+          access: 'Allow'
+          direction: 'Inbound'
+          sourceAddressPrefix: myIp
+          sourcePortRange: '*'
+          destinationAddressPrefix: '*'
+          destinationPortRange: '22'
+        }
+      }
+      {
+        name: 'HTTP'
+        properties: {
+          priority: 1010
+          protocol: 'Tcp'
+          access: 'Allow'
+          direction: 'Inbound'
+          sourceAddressPrefix: myIp
+          sourcePortRange: '*'
+          destinationAddressPrefix: '*'
+          destinationPortRange: '80'
+        }
+      }
+      {
+        name: 'HTTPS'
+        properties: {
+          priority: 1020
+          protocol: 'Tcp'
+          access: 'Allow'
+          direction: 'Inbound'
+          sourceAddressPrefix: myIp
+          sourcePortRange: '*'
+          destinationAddressPrefix: '*'
+          destinationPortRange: '443'
+        }
+      }
+    ]
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2020-05-01' = {
+  name: 'cc-vnet'
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        '10.0.0.0/22'
+      ]
+    }
+    subnets: [
+      {
+        name: 'Default'
+        properties: {
+          addressPrefix: '10.0.0.0/24'
+          networkSecurityGroup: {
+            id: nsg.id
+          }
+        }
+      }
+    ]
+  }
+}
+
+resource pip 'Microsoft.Network/publicIpAddresses@2020-05-01' = {
+  name: 'cycleserver-pip'
+  location: location
+  properties: {
+    publicIPAllocationMethod: 'Dynamic'
+  }
+}
+
+resource nic 'Microsoft.Network/networkInterfaces@2020-05-01' = {
+  name: 'cycleserver-nic'
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: '${vnet.id}/subnets/Default'
```

Comment:
Why not use symbolic name instead? https://github.com/Azure/bicep/issues/2840

Reply:
subnet.id is what we want but bicep doesn't support that... so rather than reverting to users concatenating resourceId() we should use the function.

If the linter rule throws when it's not possible (to use symbolic name) we should fix.

Reply:
Would it be better to use "existing" for the subnet and reference it that way using symbolic name?

Reply:
I'd stick with resourceId like Brian suggested - that seems to be the cleanest/most consistent current solution.  (BTW see https://github.com/Azure/bicep/issues/3886)

Reply:
I'm having a hard time following. The vnet and subnet are created in the same template above. @StephenWeatherford would you mind fixing this up in order to get this merged?

Reply:
Can't blame you. Like this:

id: resourceId('Microsoft.Network/virtualNetworks/subnets/', vnet.name, subnetName)

Reply:
Done